tutorials/baxus.ipynb

```diff
@@ -0,0 +1,985 @@
+    "\n",
+    "As mentioned above, since botorch assumes a maximization problem, we instead maximize $-f(x)$."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Define a function with dummy variables\n",
+    "\n",
+    "We first define a new function where we only pass the first two input dimensions to the actual Branin function."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "branin = Branin(negate=True).to(device=device, dtype=dtype)\n",
+    "\n",
+    "\n",
+    "def branin_emb(x):\n",
+    "    \"\"\"x is assumed to be in [-1, 1]^D\"\"\"\n",
+    "    lb, ub = branin.bounds\n",
+    "    return branin(lb + (ub - lb) * (x[..., :2] + 1) / 2)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fun = branin_emb\n",
+    "dim = 500\n",
+    "\n",
+    "n_init = 10\n",
+    "max_cholesky_size = float(\"inf\")  # Always use Cholesky"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Maintain the BAxUS state\n",
+    "BAxUS needs to maintain a state, which includes the length of the trust region, success and failure counters, success and failure tolerance, etc. \n",
+    "In contrast to TuRBO, the failure tolerance depends on the target dimensionality.\n",
+    "\n",
+    "In this tutorial we store the state in a dataclass and update the state of TuRBO after each batch evaluation. \n",
+    "\n",
+    "**Note**: These settings assume that the domain has been scaled to $[-1, 1]^d$"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@dataclass\n",
+    "class BaxusState:\n",
+    "    dim: int\n",
+    "    eval_budget: int\n",
+    "    new_bins_on_split: int = 3\n",
+    "    d_init: int = float(\"nan\")  # Note: post-initialized\n",
+    "    target_dim: int = float(\"nan\")  # Note: post-initialized\n",
+    "    n_splits: int = float(\"nan\")  # Note: post-initialized\n",
+    "    length: float = 0.8\n",
+    "    length_init: float = 0.8\n",
+    "    length_min: float = 0.5**7\n",
+    "    length_max: float = 1.6\n",
+    "    failure_counter: int = 0\n",
+    "    success_counter: int = 0\n",
+    "    success_tolerance: int = 3\n",
+    "    best_value: float = -float(\"inf\")\n",
+    "    restart_triggered: bool = False\n",
+    "\n",
+    "    def __post_init__(self):\n",
+    "        n_splits = round(math.log(self.dim, self.new_bins_on_split + 1))\n",
+    "        self.d_init = (\n",
+    "            np.argsort(\n",
+    "                [\n",
+    "                    np.abs(i * (self.new_bins_on_split + 1) ** n_splits - self.dim)\n",
+    "                    for i in range(1, self.new_bins_on_split + 1)\n",
+    "                ]\n",
+    "            )[0]\n",
+    "            + 1\n",
+    "        )\n",
```

Comment:
why not use `argmin()` instead of `argsort()[0]`?

Also, looks like there is no need to do a list comprehension here this can be vectorized

```   
        nbos = self.new_bins_on_split
        self.d_init = 1 + np.argmin(
            np.abs((1 + np.arange(nbos)) * (1 + nbos) ** n_splits - self.dim)
        )
```